static/app/components/events/interfaces/crashContent/exception/useSourceMapDebug.tsx

```diff
@@ -0,0 +1,193 @@
+  type: SourceMapProcessingIssueType.MISSING_SOURCEMAPS;
+}
+interface UrlNotValidDebugError extends BaseSourceMapDebugError {
+  data: {absValue: string};
+  type: SourceMapProcessingIssueType.URL_NOT_VALID;
+}
+interface PartialMatchDebugError extends BaseSourceMapDebugError {
+  data: {insertPath: string; matchedSourcemapPath: string};
+  type: SourceMapProcessingIssueType.PARTIAL_MATCH;
+}
+
+export type SourceMapDebugError =
+  | UnknownErrorDebugError
+  | MissingReleaseDebugError
+  | MissingUserAgentDebugError
+  | MissingSourcemapsDebugError
+  | UrlNotValidDebugError
+  | PartialMatchDebugError;
+
+export interface SourceMapDebugResponse {
+  errors: SourceMapDebugError[];
+}
+
+export enum SourceMapProcessingIssueType {
+  UNKNOWN_ERROR = 'unknown_error',
+  MISSING_RELEASE = 'no_release_on_event',
+  MISSING_USER_AGENT = 'no_user_agent_on_release',
+  MISSING_SOURCEMAPS = 'no_sourcemaps_on_release',
+  URL_NOT_VALID = 'url_not_valid',
+  PARTIAL_MATCH = 'partial_match',
+}
+
+const sourceMapDebugQuery = ({
+  orgSlug,
+  projectSlug,
+  eventId,
+  frameIdx,
+  exceptionIdx,
+}: UseSourceMapDebugProps): QueryKey => [
+  `/projects/${orgSlug}/${projectSlug}/events/${eventId}/source-map-debug/`,
+  {
+    query: {
+      frame_idx: `${frameIdx}`,
+      exception_idx: `${exceptionIdx}`,
+    },
+  },
+];
+
+interface UseSourceMapDebugProps {
+  eventId: string;
+  exceptionIdx: number;
+  frameIdx: number;
+  orgSlug: string;
+  projectSlug: string;
+}
+
+export type StacktraceFilenameQuery = {filename: string; query: UseSourceMapDebugProps};
+
+export function useSourceMapDebug(
+  props?: UseSourceMapDebugProps,
+  options: Partial<UseQueryOptions<SourceMapDebugResponse>> = {}
+) {
+  return useQuery<SourceMapDebugResponse>(props ? sourceMapDebugQuery(props) : [''], {
+    staleTime: Infinity,
+    retry: false,
+    refetchOnWindowFocus: false,
+    notifyOnChangeProps: ['data'],
+    ...options,
+    enabled: !!options.enabled && defined(props),
+  });
+}
+
+export function useSourceMapDebugQueries(props: UseSourceMapDebugProps[]) {
+  const api = useApi({persistInFlight: true});
+
+  const options = {
+    staleTime: Infinity,
+    retry: false,
+  };
+  return useQueries({
+    queries: props.map<UseQueryOptions<SourceMapDebugResponse>>(p => {
+      const key = sourceMapDebugQuery(p);
+      return {
+        queryKey: sourceMapDebugQuery(p),
+        // TODO: Move queryFn as a default in queryClient.tsx
+        queryFn: () =>
+          api.requestPromise(key[0], {
+            method: 'GET',
+            query: key[1]?.query,
+          }),
+        ...options,
+      };
+    }),
+  });
+}
+
+const ALLOWED_PLATFORMS = [
+  'node',
+  'javascript',
+  'javascript-react',
+  'javascript-angular',
+  'javascript-angularjs',
+  'javascript-backbone',
+  'javascript-ember',
+  'javascript-gatsby',
+  'javascript-vue',
+  'javascript-nextjs',
+  'javascript-remix',
+  'javascript-svelte',
```

Comment:
shall we double-check if the following platforms are not needed?

>   "javascript-browser": "javascript",
>   "javascript-capacitor": "capacitor",
>   "javascript-cordova": "cordova",
>   "javascript-electron": "electron",
>   "javascript-ionic": "ionic",
>   "javascript-wasm": "wasm"

Reply:
I noticed that although they are not listed in the following list https://github.com/getsentry/sentry/blob/6d99ee2a42feff0f727b8e33e60c115bc96081f5/static/app/data/platformCategories.tsx#L45 , they are listed in the platformicons.